src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom = parseToLocalDate(dateFrom);
+        LocalDate localDateTo = parseToLocalDate(dateTo);
+        StringBuilder result = new StringBuilder("Report for period ");
+        result.append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            int salary = 0;
+            for (String record : data) {
+                String[] splitData = record.split(" ");
+                LocalDate localDateFromData = parseToLocalDate(splitData[0]);
```

Comment:
```suggestion
                LocalDate localDateFromData = parseToLocalDate(splitData[DATE_INDEX]);
```
avoid magic numbers - use constant (especially for indexes)